inference-engine/src/transformations/src/transformations/control_flow/unroll_if.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/control_flow/unroll_if.hpp"
+
+#include <memory>
+#include <ngraph/graph_util.hpp>
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/opsets/opset6.hpp>
+#include <ngraph/opsets/opset7.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+#include <ngraph/validation_util.hpp>
+#include <vector>
+
+#include "itt.hpp"
+#include "transformations/utils/utils.hpp"
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::UnrollIf, "UnrollIf", 0);
+
+bool ngraph::pass::UnrollIf::run_on_function(std::shared_ptr<ngraph::Function> f) {
+    RUN_ON_FUNCTION_SCOPE(UnrollIf);
+    for (const auto& op : f->get_ops()) {
+        auto if_node = std::dynamic_pointer_cast<ngraph::op::v7::If>(op);
+        if (!if_node || transformation_callback(if_node)) {
+            continue;
+        }
+        Output<Node> cond {if_node->input_value(0)};
+        const auto cond_is_const = ngraph::get_constant_from_source(cond);
+        if (!cond_is_const) {
+            continue;
+        }
+        auto cond_value = cond_is_const->cast_vector<bool>();
+        auto body = (cond_value[0]) ? if_node->get_then_body() : if_node->get_else_body();
+        auto input_descriptions = if_node->get_input_descriptions(static_cast<int>(!cond_value[0]));
+        auto output_descriptions = if_node->get_output_descriptions(static_cast<int>(!cond_value[0]));
+        // connect inputs instead of body parameters
+        for (const auto& input_descr : input_descriptions) {
+            auto in_data = if_node->input_values()[input_descr->m_input_index];
```

Comment:
please check a size of `if_node->input_values()` before dereferencing using `NGRAPH_CHECK`.

Reply:
the same comment for code below